### PR TITLE
[Feature] Show error banner when video start/stop fails

### PR DESCRIPTION
**Pull request by Leah-Xia-Microsoft**
## Purpose

* Show error banner when turning on camera failed

 
 

## Does this introduce a breaking change?

```
[] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
* download the code and run it
* Mimic camera on failed error by replacing requestCameraPreviewOn and requestCameraOn functions with the following code:
```
    func requestCameraPreviewOn(state: AppState, dispatch: @escaping ActionDispatch) {
        if state.permissionState.cameraPermission == .notAsked {
            dispatch(.permissionAction(.cameraPermissionRequested))
        } else {
            callingService.requestCameraPreviewOn().map { videoStream in
                LocalUserAction.cameraOnSucceeded(videoStreamIdentifier: videoStream)
            }.sink(receiveCompletion: { completion in
                switch completion {
                case .finished:
                    break
                case .failure(let error):
                    dispatch(.localUserAction(.cameraOnFailed(error: error)))
                }
            }, receiveValue: { _ in
//                dispatch(.localUserAction(newAction))
                let error = CallCompositeInternalError.cameraOnFailed
                dispatch(.localUserAction(.cameraOnFailed(error: error)))
            }).store(in: cancelBag)
        }
    }

    func requestCameraOn(state: AppState, dispatch: @escaping ActionDispatch) {
        if state.permissionState.cameraPermission == .notAsked {
            dispatch(.permissionAction(.cameraPermissionRequested))
        } else {
            callingService.startLocalVideoStream()
                .delay(for: .seconds(1.0), scheduler: DispatchQueue.main)
                .map { videoStream in
                    LocalUserAction.cameraOnSucceeded(videoStreamIdentifier: videoStream)
                }.sink(receiveCompletion: { completion in
                    switch completion {
                    case .failure(let error):
                        dispatch(.localUserAction(.cameraOnFailed(error: error)))
                    case .finished:
                        break
                    }
                }, receiveValue: { _ in
//                    dispatch(.localUserAction(newAction))
                    let error = CallCompositeInternalError.cameraOnFailed
                    dispatch(.localUserAction(.cameraOnFailed(error: error)))
                }).store(in: cancelBag)
        }
    }
```

## What to Check
Verify that the following are valid
* Error banner shows up on both PreviewView and CallingView
* Error banner shows the right error message: "Error toggling video"

### CODE REVIEW

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Localization/en.lproj/Localizable.strings [hunk 1]
```diff
@@ -98,3 +98,4 @@
 "AzureCommunicationUICalling.SnackBar.Text.ErrorResumeCallTitle"="Unable to resume call";
 "AzureCommunicationUICalling.SnackBar.Text.ErrorResumeCallSubTitle"="Your mic is in use by another app";
 "AzureCommunicationUICalling.SnackBar.Text.ErrorCallDenied"="You were denied access to the call.";
+"AzureCommunicationUICalling.SnackBar.Text.CameraOnFailed"="Error toggling video";
```
Comment (jpeng-ms) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Localization/en.lproj/Localizable.strings:101]: Hi Leah, did we get this message from Alex?

Comment (Leah-Xia-Microsoft) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Localization/en.lproj/Localizable.strings:101]: > Hi Leah, did we get this message from Alex?

It's from the Figma design https://www.figma.com/file/PqjnN5yvNZ5EUAEBsfDQ2G/Android-%26-iOS---Call-composite-MVP0?node-id=1439%3A47467

Comment (jpeng-ms) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Localization/en.lproj/Localizable.strings:101]: Thanks!

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Tests/Presentation/Setup/ErrorInfoViewModelTests.swift [hunk 2]
```diff
@@ -83,6 +83,17 @@ class ErrorInfoViewModelTests: XCTestCase {
...
     }
 
+    func test_errorInfoViewModel_update_when_errorStateCameraOnFailedSet_then_snackBarErrorCameraOnFailedMessageDisplayed() {
+        let sut = makeSUT()
+        let state = ErrorState(internalError: .cameraOnFailed,
+                               error: nil,
+                               errorCategory: .callState)
+
+        sut.update(errorState: state)
+        XCTAssertEqual(sut.isDisplayed, true)
```
Comment (emlynmac) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Tests/Presentation/Setup/ErrorInfoViewModelTests.swift:93]: You could use an XCTAssertTrue too

**Review by kevinyulu — DISMISSED** [review]

**Review by kevinyulu — DISMISSED** [review]

**Review by emlynmac — APPROVED** [review]
LGTM

**Review by kevinyulu — APPROVED** [review]

**Pull request merged**